root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1787,3 +1776,118 @@ test('Cleanup', function (t) {
+        t.test('input URL [' + i + '] = ' + subtest.i_url, {}, function(st) {
+            var tested = false;
+            if (!subtest.i_url) {
+                st.fail('Test is invalid: Input URL "i_url" is missing: ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            if (subtest.i_ent) {
+                if (subtest.hasOwnProperty('match')) {
+                    if (previous_match_tests.indexOf(subtest.i_ent + '+' + subtest.i_url) !== -1) {
+                        st.fail('Match test is worthless: Duplication has been detected: ' + JSON.stringify(subtest));
+                    }
+                    doMatchSubtest(st, subtest.i_ent, subtest.i_url, 'input', subtest.match);
+                    tested = true;
+                } else {
+                    st.fail('Test is invalid: Input entity type "i_ent" is specified without expected relationship type "match": ' + JSON.stringify(subtest));
+                    st.end();
+                    return;
+                }
+            } else if (subtest.hasOwnProperty('match')) {
+                st.fail('Test is invalid: Expected relationship type "match" is specified without input URL "i_ent": ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            var act_clean = cleanURL(subtest.i_url);
+            if (subtest.clean) {
+                st.equal(act_clean, subtest.clean, 'Clean up');
+                if (subtest.i_ent && subtest.hasOwnProperty('match') && previous_match_tests.indexOf(subtest.i_ent + '+' + subtest.clean) === -1) {
+                    doMatchSubtest(st, subtest.i_ent, subtest.clean, 'clean', subtest.match);
+                }
+                tested = true;
+            }
+            if (subtest.i_rel && !subtest.valid && !subtest.wrong) {
+                st.fail('Test is invalid: Input relationship type "i_rel" is specified without neither "valid" nor "wrong" array: ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            if (subtest.valid && subtest.wrong) {
+                st.fail('Test is invalid: Both "valid" and "wrong" arrays are specified whereas only one can be. (One defines the other.): ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            if (subtest.valid || subtest.wrong) {
+                var match = subtest.i_rel || subtest.match;
+                var clean = subtest.clean || act_clean;
+                if (!match) {
+                    st.fail('Test is invalid: Validation test can not run without neither "match" nor input relationship type "i_rel": ' + JSON.stringify(subtest));
+                    st.end();
+                    return;
+                }
+                var nb_tested_rules = 0;
+                var validation_results = _.reduce(LINK_TYPES[match], function(results, rel_uuid, ent_name) {
```

Comment:
We use `entity_type` or `entityType` consistently throughout the codebase, rather than ent*name.